odl/tomo/data/uncompr_bin.py

```diff
@@ -0,0 +1,472 @@
+    +--------------+---------+------------------------------------------+
+    |'dshape'      |tuple    |For multi-elements: number of elements per|
+    |              |         |dimension. Optional for single elements.  |
+    +--------------+---------+------------------------------------------+
+
+    Parameters
+    ----------
+    field_list : sequence of dicts
+        Dictionaries describing the field elements, as returned by, e.g.,
+        `mrc_spec_fields`.
+    keys : dict
+        Dictionary with the following entries for the column headers in
+        the specification table:
+
+            - ``'id'``
+            - ``'byte_range'``
+            - ``'dtype'``
+            - ``'name'``
+            - ``'description'``
+
+    dtype_map : dict
+        Mapping from the data type specifiers in the specification table
+        to NumPy data types.
+
+    Returns
+    -------
+    standardized_fields : tuple of dicts
+        The standardized fields according to the above table.
+    """
+    # Parse the fields and represent them in a unfied way
+    conv_list = []
+    for row, field in enumerate(field_list):
+        new_field = {}
+
+        # Name and description: lowercase name, copy description
+        new_field['name'] = field[keys['name']].lower()
+        new_field['description'] = field[keys['description']]
+
+        # Get offset from ID range
+        num_range = field[keys['id']].split('-')
+        nstart = int(num_range[0])
+        nend = int(num_range[-1])  # 0 for range of type 3-
+
+        # Get byte range and set start
+        byte_range = field[keys['byte_range']].split('-')
+        byte_start = int(byte_range[0]) - 1
+        byte_end = int(byte_range[-1]) - 1
+        new_field['offset_bytes'] = byte_start
+
+        # Data type: transform to Numpy format and get shape from its
+        # itemsize and the byte range
+        dtype = dtype_map[field[keys['dtype']]]
+        byte_size = byte_end - byte_start + 1
+
+        if hasattr(dtype, 'itemsize') and byte_size % dtype.itemsize:
+            raise ValueError(
+                'in row {}: byte range {} not a multiple of itemsize {} '
+                'of the data type {}.'
+                ''.format(row + 1, field[keys['byte_range']], dtype.itemsize,
+                          field[keys['dtype']]))
+
+        new_field['dtype'] = dtype
+        new_field['size_bytes'] = byte_size
+        # Assuming 1d arrangement of multiple elements
+        # TODO: find way to handle 2d fields
+        if nend:
+            new_field['dshape'] = (nend - nstart + 1,)
+        elif hasattr(dtype, 'itemsize'):
+            new_field['dshape'] = (byte_size / dtype.itemsize,)
+        else:
+            new_field['dshape'] = (1,)
+
+        conv_list.append(new_field)
+
+    return tuple(conv_list)
+
+
+class FileReaderUncompressedBinary(object):
+
+    """Reader for uncompressed binary files including a header."""
```

Comment:
> The binary format used by EMRecon, where they give n data points and then a time stamp, i.e.

Okay, I'll add **one** header and **one** block of data.

> Non-numpy based data:

The reader does in no way rely on Numpy's format. Depending on how it's pickled it actually is supported. Numpy's `.npy` format happens to be a header-followed-by-data-block format, and that is supported by the reader.

> Formats where the interpretation depends on the header

Supported. Actually, that's exactly what happens in MRC. It has a `mode` header entry which specifies how the data is to be understood, i.e. the data type.

> Formats where each object uses more than one line

I'm not sure what you mean by that. The reader is not line-based.

> Formats where there are sub-headers for each object

Okay, that's the same case as in the first example.

> I could probably go on, but I'm just trying to say that there are heaps of formats that fit the name of this class that do not work with it.

With the "one header, one data block" restriction, what is left that isn't supported?

Regarding the name, I'm not sure how to make it adequate and not ridiculously long at the same time. `FileReaderSingleHeaderUncompressedBinaryData`?